### PR TITLE
Bookstore: connect to API

### PR DESCRIPTION
**Pull request by bobb-Rob**
## I Implemented the following:
- I used Axios with createAsyncThunk for making API calls.
- Fetched book list from Bookstore API on Books page load in the below order: 
        - Added an Action Creator, that returns a function.
        - Fetch list of books.
        - Dispatch another action (add an action type) that will pass API response to reducer and update the state.
- Refactored the add book and remove book features to persist changes in the server.
        - Changed the existing Action Creators, so they return functions.
        - Made changes by using API.
        - Dispatched an action that will updates the state accordingly.
- Add book and remove book is working the same way after the refactor.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,36 +1,63 @@
+/* eslint-disable no-restricted-syntax */
```
Comment (FrederickMih) [inline src/redux/books/books.js]: - [x] kindly fix Linters issues. Linters flag programming bugs so we need to fix them.

File: src/redux/books/books.js [hunk 2]
```diff
@@ -1,36 +1,63 @@
...
+
+export const fetchBookList = createAsyncThunk(
+  BOOKS,
+  async () => {
+    const response = await axios.get(url());
+    const bookList = await response.data;
+    const bookArray = reOrganizeData(bookList);
+    return bookArray;
   },
+);
```
Comment (FrederickMih) [inline src/redux/books/books.js]: - [x] Based on the requirements of this project, the action creator for fetching books should automatically dispatch an action type that will add a book to the books list and update the state. Kindly fix this. Let's equally do the same for the action creator for removing a book.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @bobb-Rob , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



# To Highlights

- PR is professional :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by gemmen29 — APPROVED** [review]
Hi @bobb-Rob, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**